src/lib/storage/dictionary_column.hpp

```diff
@@ -21,51 +25,115 @@ constexpr ValueID INVALID_VALUE_ID{std::numeric_limits<ValueID::base_type>::max(
 template <typename T>
 class DictionaryColumn : public BaseColumn {
  public:
+
+  DictionaryColumn() = delete;
+
   /**
    * Creates a Dictionary column from a given value column.
    */
-  explicit DictionaryColumn(const std::shared_ptr<BaseColumn>& base_column);
+  explicit DictionaryColumn(const std::shared_ptr<BaseColumn>& base_column) : _dictionary(nullptr), _attribute_vector(nullptr) {
```

Comment:
Implementation should be in an extra cpp file.

Reply:
a comment by the instructors says it should be done within this file (lines 63, 64)

i agree though, this should be in a seprate cpp file. maybe we can use `EXPLICITYLY_INSTATIATE_COLUMN_TYPES` like in `value_column.cpp`